File: 234.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        List<Integer> list = new ArrayList<Integer>();
        for (ListNode temp = head; temp != null; temp = temp.next) {
            list.add(temp.val);
        }
        System.out.print(list);
        for (int i = 0; i < list.size()/2; i++) {
            if (!list.get(i).equals(list.get(list.size()-i-1))) {
                return false;
            }
        }
        return true;
    }
}

File: 234.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def isPalindrome(self, head):
        """
        :type head: ListNode
        :rtype: bool
        """
        temp = []
        while head:
            temp.append(head.val)
            head = head.next
        for i in range(len(temp)/2):
            if temp[i] != temp[len(temp)-i-1]:
                return False
        return True